### ISSUE TITLE
Rejection object is empty on response error in interceptor

### ISSUE THREAD

**Issue by simonjodet** [angular/angular.js#10870]
As you can see in the capture below, I get a (expected) 401 error from the server but the rejection object in the responseError of the interceptor is empty (attaching interceptor's code below). Am I doing something wrong or is it a bug?

![developer_tools_-_http___127_0_0_1_9000_](https://cloud.githubusercontent.com/assets/460700/5898381/3ecbc960-a54f-11e4-9930-b169a43529f8.jpg)

```javascript
$provide.factory('xhrInterceptor', ['$q', '$rootScope', '$log', '$location', '$cookieStore', function($q, $rootScope, $log, $location) {
            $rootScope.ongoingXhrRequestsCount = 0;
            return {
                'request': function(config) {
                    $rootScope.ongoingXhrRequestsCount = $rootScope.ongoingXhrRequestsCount + 1;
                    return config || $q.when(config);
                },
                'requestError': function(rejection) {
                    $rootScope.ongoingXhrRequestsCount = $rootScope.ongoingXhrRequestsCount - 1;
                    return $q.reject(rejection);
                },
                'response': function(response) {
                    $rootScope.ongoingXhrRequestsCount = $rootScope.ongoingXhrRequestsCount - 1;
                    return response || $q.when(response);
                },
                'responseError': function(rejection) {
                    $rootScope.ongoingXhrRequestsCount = $rootScope.ongoingXhrRequestsCount - 1;
                    if (rejection.status == 401) {
                        $location.path('/login');
                        location.reload();
                    }
                    else {
                        var body;
                        if (rejection && rejection.data && rejection.data.body) {
                            body = rejection.data.body;
                        }
                        $log.error('Error returned by API.', 'Status: ', rejection.status, '. Body: ', body, '. Rejection: ', rejection);
                    }
                    return $q.reject(rejection);
                }
            };
        }]);
```

**Comment by pkozlowski-opensource** [comment]
@simonjodet hard to say based only on the snippet / picture provided. Could you please put a live reproduce scenario using plunker or similar?

BTW: you don't need to count pending requests in your interceptor, this info is already available in `pendingRequests` of $http: https://docs.angularjs.org/api/ng/service/$http

**Comment by pkozlowski-opensource** [comment]
BTW, looking at the screenshot it might be a CORS issue and it his case you would be getting status 0, potentially....

**Comment by simonjodet** [comment]
Hi Pawel,

Your two messages helped me figure it out.
When I tried doing a plunker using https://httpbin.org/status/401 instead of my API, I couldn't reproduce the issue.

In the end, my API was sending CORS headers (such as Access-Control-Allow-Headers and Access-Control-Allow-Origin) on OPTIONS requests but not on other methods requests.

I added them and it works fine now. So closing this issue and thanks for the help.

Thanks!
